homeassistant/components/modbus/switch.py

```diff
@@ -111,11 +116,10 @@ async def async_turn_on(self, **kwargs):
         result = await self._hub.async_pymodbus_call(
             self._slave, self._address, self._command_on, self._write_type
         )
-        if result is False:
+        if result is None:
             self._available = False
             self.async_write_ha_state()
         else:
-            self._available = True
```

Comment:
Should available only be set `True` by `async_update` and not by a service handler method (turn_on/turn off)?

Reply:
We need to set _available to true if "verify:" isn't configured, because in that case async_update might not be called for a long time (many users set scan_interval: 9999, because the want to avoid the polling).

Reply:
Right, but we could do that already in the service handler method?

Reply:
I do not follow you, I cannot do it in modbus.py because setting it in async_pymodbus_call would mean all platforms on that hub are unavailable. Setting it in the platform means only that platform is unavailable.

Right now async_pymodbus_call() informs the platform that the call failed (return none) and the platform makes itself and only itself unavailable.

Reply:
If the return value is not None we could set available to True like we do in the other platforms.

Reply:
DONE, the argument for not having it was that you can only switch if _available is true.